src/ServiceControl.UnitTests/Archiving/ArchiveGroupTests.cs

```diff
@@ -65,5 +65,60 @@ await handler.Handle(message, context)
                 }
             }
         }
+
+        [Test]
+        public async Task ArchiveGroup_GetGroupDetails_doesnt_fail_with_invalid_groupId()
+        {
+            // Arrange
+            using (var documentStore = InMemoryStoreBuilder.GetInMemoryStore())
+            {
+                var failureGroupsViewIndex = new FailureGroupsViewIndex();
+                await failureGroupsViewIndex.ExecuteAsync(documentStore);
+
+                var groupId = "TestGroup";
+                var previousArchiveBatchId = ArchiveBatch.MakeId(groupId, ArchiveType.FailureGroup, 1);
+
+                using (var session = documentStore.OpenAsyncSession())
+                {
+                    var previousAchiveBatch = new ArchiveBatch { Id = previousArchiveBatchId };
+                    await session.StoreAsync(previousAchiveBatch)
+                        .ConfigureAwait(false);
+
+                    var previousArchiveOperation = new ArchiveOperation
+                    {
+                        Id = ArchiveOperation.MakeId(groupId, ArchiveType.FailureGroup),
+                        RequestId = groupId,
+                        ArchiveType = ArchiveType.FailureGroup,
+                        TotalNumberOfMessages = 2,
+                        NumberOfMessagesArchived = 0,
+                        Started = DateTime.Now,
+                        GroupName = "Test Group",
+                        NumberOfBatches = 3,
+                        CurrentBatch = 0
+                    };
+                    await session.StoreAsync(previousArchiveOperation)
+                        .ConfigureAwait(false);
+
+                    await session.SaveChangesAsync()
+                        .ConfigureAwait(false);
+                }
+
+                var documentManager = new ArchiveDocumentManager();
+                var domainEvents = new FakeDomainEvents();
+                var archivingManager = new ArchivingManager(domainEvents);
+                var retryingManager = new RetryingManager(domainEvents);
+                var handler = new ArchiveAllInGroupHandler(documentStore, domainEvents, documentManager, archivingManager, retryingManager);
```

Comment:
Given that we only care about the handler I would tend to inline the things than can be inlined

Reply:
Inlined. Also did it for the other test